lib/components/home/artist_rails/artist_rail.js

```diff
@@ -30,35 +39,88 @@ class ArtistRail extends React.Component {
 
   componentWillReceiveProps(nextProps) {
     if (nextProps.rail.results) {
-      this.setState({ artists: nextProps.rail.results })
+      const artists = nextProps.rail.results
+      artists.forEach((artist) => {
+        artist._animatedValues = {
+          opacity: new Animated.Value(1),
+          translateY: new Animated.Value(0),
+        }
+      })
+      this.setState({ artists })
     }
   }
 
-  /**
-   * Removes the followed artist from the list and re-renders.
-   */
-  handleFollowChange(artist) {
-    metaphysics(suggestedArtistQuery(artist._id))
-      .then(({ me: { suggested_artists }}) => suggested_artists[0]) // return artist or undefined
-      .catch(error => console.error(error))                         // return undefined
-      .then((suggested_artist) => {
-        const artists = this.state.artists.slice(0)                 // copy array
-        const index = artists.indexOf(artist)
-        if (suggested_artist) {
-          artists[index] = suggested_artist                         // replace card with new suggestion
-        } else {
-          artists.splice(index, 1)                                  // remove card when no suggestion or on error
-        }
-        this.setState({ artists })
-      })
+  followedArtistAnimation(followedArtist) {
+    return new Promise((resolve, reject) => {
+      const { opacity, translateY } = followedArtist._animatedValues
+      const duration = Animation.duration.followedArtist
+      const easing = Animation.easing
+      Animated.parallel([
+        Animated.timing(opacity,    { duration, easing, toValue: 0 }),
+        Animated.timing(translateY, { duration, easing, toValue: Animation.yDelta }),
+      ]).start(resolve)
+    })
+  }
+
+  suggestedArtistAnimation(suggestedArtist) {
+    return new Promise((resolve, reject) => {
+      const { opacity, translateY } = suggestedArtist._animatedValues
+      const duration = Animation.duration.suggestedArtist
+      const easing = Animation.easing
+      Animated.parallel([
+        Animated.timing(opacity,    { duration, easing, toValue: 1 }),
+        Animated.timing(translateY, { duration, easing, toValue: 0 }),
+      ]).start(resolve)
+    })
+  }
+
+  replaceFollowedArtist(followedArtist, suggestedArtist) {
+    const artists = this.state.artists.slice(0)
+    const index = artists.indexOf(followedArtist)
+    if (suggestedArtist) {
+      suggestedArtist._animatedValues = {
+        opacity: new Animated.Value(0),
+        translateY: new Animated.Value(-Animation.yDelta),
+      }
+      artists[index] = suggestedArtist
+    } else {
+      // remove card when there is no suggestion
+      artists.splice(index, 1)
+    }
+    this.setState({ artists })
+    return suggestedArtist
+  }
+
+  handleFollowChange(followedArtist, setFollowButtonStatus) {
+      // Get a new suggested artist based on the followed artist.
+    return metaphysics(suggestedArtistQuery(followedArtist._id))
```

Comment:
No because in my earlier attempt I was wrapping the previous and new suggestions in a single component, I would then re-render the component with a clean state after the animation finished, the re-render was what triggered it.